factory/src/main/kotlin/io/qalipsis/core/factory/meters/CompositeGauge.kt

```diff
@@ -1,43 +1,56 @@
-package io.qalipsis.core.factory.meters
-
-import io.qalipsis.api.meters.Gauge
-import io.qalipsis.api.meters.Meter
-
-/**
- * Composite class to encapsulate the [Gauge]s at the scenario and campaign
- * level in order to update both at the same time.
+/*
+ * QALIPSIS
+ * Copyright (C) 2024 AERIS IT Solutions GmbH
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
  *
- * This [Gauge] should be seen as the [scenarioLevelGauge] by the calling application.
- * The meter registries are the only one aware of the existence of the [campaignLevelGauge]
- * and will ask for its publication when required.
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
  *
- * This instance of [Gauge] is not known by the instances of Qalipsis measurement publisher.
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
  *
- * @author Joël Valère
  */
-internal data class CompositeGauge(
-    private val scenarioLevelGauge: Gauge,
-    private val campaignLevelGauge: Gauge
-) : Gauge by scenarioLevelGauge {
-
-    override val id: Meter.Id
-        get() = scenarioLevelGauge.id
-
-    override fun report(configure: Meter.ReportingConfiguration<Gauge>.() -> Unit): Gauge =
-        scenarioLevelGauge.report(configure)
 
-    fun toByte() = scenarioLevelGauge.value().toInt().toByte()
-
-    fun toChar(): Char = scenarioLevelGauge.value().toInt().toChar()
-
-    fun toDouble() = scenarioLevelGauge.value()
-
-    fun toFloat() = scenarioLevelGauge.value().toFloat()
-
-    fun toInt() = scenarioLevelGauge.value().toInt()
+package io.qalipsis.core.factory.meters
 
-    fun toLong() = scenarioLevelGauge.value().toLong()
+import io.qalipsis.api.meters.Gauge
+import io.qalipsis.api.meters.Meter
 
-    fun toShort() = scenarioLevelGauge.value().toInt().toShort()
+/**
+ * Implementation of [Gauge] that updates the scenario relevant meter [scenarioMeter] but also
+ * the meter for the whole campaign level [globalMeter].
```

Comment:
Just a suggestion
```suggestion
 * Implementation of [Gauge] that updates the scenario-relevant meter [scenarioMeter], as well as
 * the meter at the global campaign level [globalMeter].
```
Other docs similar to this can be editited into this.

Reply:
There are still some other documentation on the meters similar to the aforementioned and the statement sounds incomplete. I think this is what you were trying to write
``` 
Implementation of [Gauge] that updates not only the scenario relevant meter [scenarioMeter] but also
the meter for the whole campaign level [globalMeter].
 ```
 
 I think all the other docs on the other meters should be updated.

Reply:
Ohh all? I thought it was just this.
Sorry about that